src/Application/Common/Extensions/ApplicationCacheServiceExtension.cs

```diff
@@ -0,0 +1,38 @@
+﻿using CleanArchitecture.Application.Common.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CleanArchitecture.Application.Common.Extensions
+{
+    public static class ApplicationCacheServiceExtension
+    {
+        public static T GetAndSet<T>(this IApplicationCacheService applicationCacheService, string key, Func<T> getResult, TimeSpan expireTime)
+        {
+            var data = applicationCacheService.Get<T>(key);
+
+            if (data == null)
+            {
+                data = getResult();
+                applicationCacheService.Set(key, data, expireTime);
+            }
+
+            return data;
+        }
+
+        public static Task<T> GetAndSetAsync<T>(this IApplicationCacheService applicationCacheService, string key, Func<Task<T>> getResult, TimeSpan expireTime)
+        {
+            var data = applicationCacheService.Get<T>(key);
+
+            if (data == null)
+            {
+                data = getResult().GetAwaiter().GetResult();
```

Comment:
Why not using async/await here?

Reply:
Can be use but GetAwaiter extension already get awaitable object. @deceptiveSimplicity

Reply:
`.GetAwaiter().GetResult()` will wait synchronous, blocking the current thread. That's why I am asking.

Reply:
You are right. Can you check commit 048a72b4e5aa922d6ab34f96297c07b3a9b4ed96 @deceptiveSimplicity ?